lib/shopify-cli/shopifolk_check.rb

```diff
@@ -0,0 +1,57 @@
+require 'fileutils'
+
+module ShopifyCli
+  class ShopifolkCheck
+    def self.assume_shopifolk?
+      ShopifyCli::ShopifolkCheck.new.assume_shopifolk?
+    end
+
+    def assume_shopifolk?
+      is_shopifolk = if shopifolk_by_dev? && shopifolk_by_feature?
+        true
+      else
+        false
+      end
+      is_shopifolk
+    end
+
+    def shopifolk_by_dev?
+      if File.exist?('/opt/dev/bin/dev') && File.exist?('/opt/dev/.shopify-build')
+        true
+      else
+        false
+      end
+    end
+
+    def shopifolk_by_feature?
+      set_shopifolk_flag_by_gcloud_config
```

Comment:
There's a useful principle called [Command-Query Seperation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation) which says that a method should return information, or perform an action, but not both. Here it seems we have a 'side-effect' since we call `set_shopifolk_flag_by_gcloud_config` before return a value. Can you see a way to restructure things to avoid that?

Reply:
I see! Would a solution be to call `set_shopifolk_flag_by_gcloud_config` in `assume_shopifolk?` instead of `shopifolk_by_feature?`?